hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/UsingNuma.md

```diff
@@ -0,0 +1,180 @@
+from the below configurations.
+
+In `yarn-site.xml` add
+
+```
+    <property>
+        <name>yarn.nodemanager.numa-awareness.read-topology</name>
+        <value>false</value>
+    </property>
+```
+
+**3) Numa command**
+
+This property is passed when `yarn.nodemanager.numa-awareness.read-topology` is set to true.
+It is recommended to verify the installation of `numactl` command in the Linux OS of every node.
+
+Use `/usr/bin/numactl --hardware` to verify.
+
+In `yarn-site.xml` add
+
+```
+    <property>
+         <name>yarn.nodemanager.numa-awareness.numactl.cmd</name>
+         <value>/usr/bin/numactl</value>
+    </property>
+```
+
+**4) NUMA nodes id’s**
+
+This property is used to provide the NUMA node ids as comma separated values.It will be read only
+when the `yarn.nodemanager.numa-awareness.read-topology` is false.
+
+In ```yarn-site.xml``` add
+
+```
+    <property>
+        <name>yarn.nodemanager.numa-awareness.node-ids</name>
+        <value>0,1</value>
+    </property>
+```
+
+**5) NUMA Node memory**
+
+This property will be used to read the memory(in MB) configured for each NUMA node specified in
+`yarn.nodemanager.numa-awareness.node-ids` by substituting the node id in the place of
+`<NODE_ID>`.It will be read only when the `yarn.nodemanager.numa-awareness.read-topology`
+is false.
+
+In ```yarn-site.xml``` add
+
+```
+    <property>
+        <name>yarn.nodemanager.numa-awareness.<NODE_ID>.memory</name>
+        <value>8192</value>
+    </property>
+```
+
+**6) NUMA Node CPUs**
+
+This property will be used to read the number of CPUs configured for each node specified in
+`yarn.nodemanager.numa-awareness.node-ids` by substituting the node id in the place of
+`<NODE_ID>`.It will be read only when the `yarn.nodemanager.numa-awareness.read-topology` is false.
+
+In ```yarn-site.xml``` add
+
+```
+    <property>
+        <name>yarn.nodemanager.numa-awareness.<NODE_ID>.cpus</name>
+        <value>8192</value>
+    </property>
+```
+
+**7) Passing java_opts for map/reduce**
+
+Every container has to be aware of NUMA and the JVM can be notified via passing NUMA flag.
+
+In ```mapred-site.xml``` add
+
+```
+    <property>
+        <name>mapreduce.reduce.java.opts</name>
+        <value>-XX:+UseNUMA</value>
+    </property>
+    <property>
+        <name>mapreduce.map.java.opts</name>
+        <value>-XX:+UseNUMA</value>
+    </property>
+```
+
+# Default configuration
+
+| Property | Default value |
+| --- |-----|
+|yarn.nodemanager.numa-awareness.enabled|false|
+|yarn.nodemanager.numa-awareness.read-topology|false|
+
+# Enable numa balancing at OS Level (Optional)
+
+In linux, by default numa balancing is by default off. For more performance improvement,
+NumaBalancing can be turned on for all the nodes in cluster
+
+```
+echo 1 | sudo tee /proc/sys/kernel/numa_balancing
+```
+
+# Verify
+
+**1) NameNode log**
+
+In any of the namenode, grep the logs with command
```

Comment:
NodeManager instead of namenode.

grep the logs => grep the nodemanager log file using below command